src/ccs/lcccs.rs

```diff
@@ -110,64 +108,132 @@ impl<C: CurveGroup> LCCCS<C> {
     }
 
     pub fn fold(
-        lcccs1: &Self,
-        cccs2: &CCCS<C>,
-        sigmas: &[C::ScalarField],
-        thetas: &[C::ScalarField],
+        lcccs1: &[Self],
+        cccs2: &[CCCS<C>],
+        sigmas: &[Vec<C::ScalarField>],
+        thetas: &[Vec<C::ScalarField>],
         r_x_prime: Vec<C::ScalarField>,
         rho: C::ScalarField,
     ) -> Self {
-        let C = Commitment(lcccs1.C.0 + cccs2.C.0.mul(rho));
-        let u = lcccs1.u + rho;
-        let x: Vec<C::ScalarField> = lcccs1
-            .x
-            .iter()
-            .zip(
-                cccs2
-                    .x
-                    .iter()
-                    .map(|x_i| *x_i * rho)
-                    .collect::<Vec<C::ScalarField>>(),
-            )
-            .map(|(a_i, b_i)| *a_i + b_i)
-            .collect();
-        let v: Vec<C::ScalarField> = sigmas
-            .iter()
-            .zip(
-                thetas
-                    .iter()
-                    .map(|x_i| *x_i * rho)
-                    .collect::<Vec<C::ScalarField>>(),
-            )
-            .map(|(a_i, b_i)| *a_i + b_i)
-            .collect();
+        let mut C_folded = lcccs1[0].C.0;
+        let mut u_folded = lcccs1[0].u;
+        let mut x_folded: Vec<C::ScalarField> = lcccs1[0].x.clone();
+        let mut v_folded: Vec<C::ScalarField> = sigmas[0].clone();
```

Comment:
Slightly confused about what you are doing here. Can you document?

Reply:
Right, I've updated the whole method and got rid of this part.